python/rmm/tests/test_rmm.py

```diff
@@ -765,3 +765,70 @@ def deallocate_func(ptr, size):
 
     captured = capsys.readouterr()
     assert captured.out == "Allocating 256 bytes\nDeallocating 256 bytes\n"
+
+
+@pytest.fixture
+def make_reinit_hook():
+    funcs = []
+
+    def _make_reinit_hook(func, *args, **kwargs):
+        funcs.append(func)
+        rmm.register_reinitialize_hook(func, *args, **kwargs)
+        return func
+
+    yield _make_reinit_hook
+    for func in funcs:
+        rmm.unregister_reinitialize_hook(func)
+
+
+def test_reinit_hooks_register(make_reinit_hook, capsys):
+    make_reinit_hook(lambda: print("one"))
+    make_reinit_hook(lambda: print("two"))
+    make_reinit_hook(lambda x: print(x), "three")
```

Comment:
Instead of using `sys.stdout` what if we just have a `list` (say `L`) used by the lambdas and then we just do `L.append`. May be less brittle

Note: This shows up in a few other places, but just commenting on this instance

Reply:
Fix'd to use a list instead!